### ISSUE TITLE
JIT Exporter for Transformer Model

### ISSUE THREAD

**Issue by gvskalyan** [pytorch/fairseq#873]
Is there any update on using TorchScript annotations (nn.Module->torch.jit.ScriptModule, script_method and trace) to load transformer model without using Python interpretor and end-end inference including beam search.

**Comment by huihuifan** [comment]
@myleott?

**Comment by gvskalyan** [comment]
Something like this end-end  https://twitter.com/Thom_Wolf/status/1151169470498582529

**Comment by gvskalyan** [comment]
@zhangguanheng66 I think transformer is jit traceable and does it decrease latency (in case of transformer) after being converted.

**Comment by zhangguanheng66** [comment]
@gvskalyan Yes. the transformer module in pytorch core library is jit traceable, which should decrease the latency. But I haven't benchmarked it yet.

**Comment by gvskalyan** [comment]
An another fork, https://github.com/NVIDIA/DeepLearningExamples/tree/master/PyTorch/Translation/Transformer#changelog
Please refer June 2019 in changelog - jit support added

**Comment by Meteorix** [comment]
@gvskalyan @zhangguanheng66 I have tried [bert jit](https://github.com/huggingface/pytorch-transformers/blob/master/docs/source/torchscript.rst) and benchmarked it. It will be 25% faster on gpu.

**Comment by Meteorix** [comment]
Still working on fairseq jit. Beam search may be a big issue.

**Comment by Oktai15** [comment]
@Meteorix do you know where one may find simple jit-transformer with beam search?

**Comment by Meteorix** [comment]
> @Meteorix do you know where one may find simple jit-transformer with beam search?

https://github.com/pytorch/translate I used this repo a couple of months ago.

**Comment by Oktai15** [comment]
@Meteorix have you seen jittable LM with beam search?

**Comment by Gabber** [comment]
Hi guys, any news on this one?
